draft-ietf-sframe-enc.md

```diff
@@ -311,28 +303,61 @@ Key or Key Length (K, 3 bits):
 : This field contains the key id (KID) if the X flag is set to 0, or the key
 length (KLEN) if set to 1.
 
-If X flag is 0, then the KID is in the range of 0-7 and the counter (CTR) is
-found in the next LEN bytes:
+Extended Counter Flag (Y, 1 bit):
+: Indicates if the key field contains the key id or the key length.
 
-~~~~~ aasvg
- 0 1 2 3 4 5 6 7
-+-+-----+-+-----+---------------------------------+
-|R|LEN  |0| KID |    CTR... (length=LEN)          |
-+-+-----+-+-----+---------------------------------+
-~~~~~
-{: title="SFrame header with short KID" }
+Counter or Counter Length (C, 3 bits):
+: This field contains the key id (KID) if the X flag is set to 0, or the key
+length (KLEN) if set to 1.
 
-If X flag is 1 then KLEN is the length of the key (KID) in bytes, minus one
-(the range of possible lengths is thus 1-8). The KID is encoded in
-the KLEN bytes following the metadata byte, and the counter (CTR) is encoded
-in the next LEN bytes:
+The Key ID and Counter fields are encoded as compact unsigned integers in
+network (big-endian) byte order.  If the value of one of these fields is in the
+range 0-7, then the value is carried in the corresponding bits of the config
+byte (K or C) and the corresponding flag (X or Y) is set to zero.  Otherwise,
+the value is MUST be encoded with the minimum number of bytes required and
```

Comment:
Interim 2023-09-13:
* We should be clear that receivers MUST reject frames that are not minimally encoded

```suggestion
the value MUST be encoded with the minimum number of bytes required and
```